src/main/scala/edu/twitter/evaluation/ModelEvaluator.scala

```diff
@@ -0,0 +1,116 @@
+package edu.twitter.evaluation
+
+import edu.twitter.model.{GradientBoostingModel, TweetsLoader}
+import org.apache.spark.SparkContext
+import org.apache.spark.mllib.feature.HashingTF
+import org.apache.spark.rdd.RDD
+import org.elasticsearch.spark.rdd.EsSpark
+
+/** Grouping of the evaluation necessary fields
+  * these fields are used for evaluating the training error. */
+case class EvaluationFields(happyCorrect: Int, happyTotal: Int, sadCorrect: Int, sadTotal: Int) {
+
+  /**
+    * Combining two evaluation instances.
+    *
+    * @param o instance to combine with this one.
+    * @return a new instance resulting of the combination of
+    *         both instances.
+    */
+  def +(o: EvaluationFields): EvaluationFields = o match {
+    case EvaluationFields(oHappyCorrect, oHappyTotal, oSadCorrect, oSadTotal) =>
+      EvaluationFields(
+        happyCorrect + oHappyCorrect,
+        happyTotal + oHappyTotal,
+        sadCorrect + oSadCorrect,
+        sadTotal + oSadTotal
+      )
+  }
+}
+
+/** Representation of a training tweet that has been classified
+  * by the model, it holds both the actual and model labels. */
+case class EvaluatedTrainingTweet(actualLabel: Double, modelPrediction: Double, tweetText: String)
+
+/**
+  * Responsible for evaluating a model based on a given
+  * testing data. this class will print training analysis
+  * in the console and persist the labeled data in `Elasticsearch`
+  * for further visualization.
+  *
+  * @param sc spark context.
+  */
+class ModelEvaluator(sc: SparkContext) {
+
+  /**
+    * Given a model will evaluate it based on
+    * training data and log evaluation analysis.
+    *
+    * @param model target model for evaluation.
+    */
+  def evaluate(model: GradientBoostingModel#GenericModel): Unit = {
+    val evaluation = evaluateData(model)
+    performEvaluationAnalysis(evaluation)
+  }
+
+  /**
+    * Given a model will evaluate it based on
+    * training data, log evaluation analysis and
+    * persist the labeled records in `Elasticsearch`
+    *
+    * @param model target model for evaluation
+    */
+  def evaluateAndPersist(model: GradientBoostingModel#GenericModel): Unit = {
+    val evaluation = evaluateData(model)
+    performEvaluationAnalysis(evaluation)
+    EsSpark.saveToEs(evaluation, "training-analysis")
+  }
+
+  /**
+    * Evaluate all the testing data.
+    *
+    * @param model target model for evaluation
+    * @return rdd of `EvaluatedTrainingTweet` instances.
+    */
+  private def evaluateData(model: GradientBoostingModel#GenericModel): RDD[EvaluatedTrainingTweet] = {
```

Comment:
We couldn't simply load all data we need to separate training error from the testing error

Reply:
Good point, I was going to add a method just for that. we will provide it with training and testing data